lib/java/com/google/android/material/tabs/TabLayout.java

```diff
@@ -2852,14 +2852,16 @@ private void clipViewToPaddingForBadge(boolean flag) {
     final void updateOrientation() {
       setOrientation(inlineLabel ? HORIZONTAL : VERTICAL);
       if (customTextView != null || customIconView != null) {
-        updateTextAndIcon(customTextView, customIconView);
+        updateTextAndIcon(customTextView, customIconView, /* isDefaultTabView */ false);
       } else {
-        updateTextAndIcon(textView, iconView);
+        updateTextAndIcon(textView, iconView, /* isDefaultTabView */ true);
       }
     }
 
     private void updateTextAndIcon(
-        @Nullable final TextView textView, @Nullable final ImageView iconView) {
+        @Nullable final TextView textView,
+        @Nullable final ImageView iconView,
+        final boolean isDefaultTabView) {
```

Comment:
nit) I prefer to make the method more context-free. How about renaming this to something like addIconMargins?